modules/data_release/php/data_release.class.inc

```diff
@@ -51,70 +54,83 @@ class Data_Release extends \NDB_Menu_Filter
             ON (dr.id=drp.data_release_id) 
             JOIN users u ON (u.ID=drp.userid) 
             WHERE u.UserID="
-            . $DB->quote($user->getUsername());
+                . $DB->quote($user->getUsername());
         }
 
-        $this->group_by     = '';
-        $this->order_by     = 'upload_date';
-        $this->headers      = array(
-                               'file_name',
-                               'version',
-                               'upload_date',
-                              );
-        $this->validFilters = array(
-                               'file_name',
-                               'version',
-                               'upload_date',
-                              );
-        $this->formToFilter = array(
-                               'file_name'   => 'file_name',
-                               'version'     => 'version',
-                               'upload_date' => 'upload_date',
-                              );
-
-        $userUpload     = $user->hasPermission("data_release_upload");
-        $userEditAccess = $user->hasPermission("data_release_edit_file_access");
+        $this->group_by = '';
+        $this->order_by = 'uploadDate';
+    }
 
-        $userids = $DB->pselectColWithIndexKey(
-            "SELECT ID, UserID FROM users",
-            array(),
-            "ID"
-        );
+    /**
+     * Create the form for the data release menu page
+     *
+     * @return void
+     **/
+    function setup()
+    {
+        parent::setup();
 
-        $versionedFiles        = $this->getVersionedFiles($DB);
-        $data_release_ids      = array();
-        $data_release_versions = array();
+        // Get list of users in the database
+        $list_of_users = $this->getUsersList();
 
-        foreach ($versionedFiles as $versionName=>$versionFiles) {
-            foreach ($versionFiles as $fileID=>$fileName) {
-                $data_release_ids[$fileID] = $fileName . " - " . $versionName;
-            }
-            $data_release_versions[$versionName] = $versionName;
+        // Get list of data release versions and filenames
+        $versionedFiles = $this->getVersionedFiles();
 
+        $filenames_options = array();
+        $versions_options  = array();
+
+        foreach ($versionedFiles as $versionName => $versionFiles) {
```

Comment:
Not a blocking change but would you mind changing these variable names? I know you didn't write them in the first place but it looks like it's iterating over the same array in both loops because the names are odd.

Reply:
how about I rename `$versionFiles` to `$filesArray`? Would that be less confusing?

I could also rename `$versionedFiles` to `$filesOrganizedPerDataReleaseVersions`?

Any better idea?

Reply:
`$vFiles` is used instead of `$versionedFiles` elsewhere in this module so maybe that?

I don't really mind as long as the words look less similar. Thinking of last week's clean code ADM :)

Reply:
Done! Thank you!